main.py

```diff
@@ -0,0 +1,69 @@
+"""Proxy server to communicate a client to ARTIQ."""
+
+from contextlib import asynccontextmanager
+import json
+import os
+
+from fastapi import FastAPI
+from sipyco.pc_rpc import Client
+
+
+settings = {}
+
+
+def load_setup_file():
+    """Loads set-up information from the setup file.
+
+    The file should have the following JSON structure:
+
+      {
+        "master_path": {master_path}
+      }
+    """
+    global settings
+    with open("setup.json", encoding="utf-8") as setup_file:
+        settings = json.load(setup_file)
+
+
+@asynccontextmanager
+async def lifespan(_app: FastAPI):
+    """Lifespan events.
+
+    This function is set as the lifespan of the application.
+
+    The code before 'yield' is executed once before the application starts.
+    The code after 'yield' is executed when the application is shutting down.
+    """
+    load_setup_file()
+    yield
+
+
+app = FastAPI(lifespan=lifespan)
+
+
+@app.get("/ls/")
+async def list_directory(directory: str = ""):
+    """Get the list of elements in the given path.
+
+    The 'master_path' is used for the prefix of the path.
+    """
+    remote = get_client("master_experiment_db")
+    contents = remote.list_directory(os.path.join(settings["master_path"], directory))
```

Comment:
Could you test if the `AsyncioClient` works as well?
Since we defined this as a coroutine, we can use `await` here.

Even though the FastAPI will run it asynchronously, I think we can practice to use `asyncio`!

In the [reference](https://fastapi.tiangolo.com/async/#in-a-hurry) you provided:

> Anyway, in any of the cases above, FastAPI will still work asynchronously and be extremely fast.

Reply:
I didn't analyze for the `AsyncioClient`.. And also `artiq_client.py` doesn't use it.
In `artiq_dashboard.py`, it uses both `Client` and `AsyncioClient` so if we need, we can look at it.

If you think we should change into `asyncio`, I'd like to make it as another issue since this proxy server causes a bottlenect of `iquip` project.

Reply:
Oh, okay then. Let's use `Client` for now.